models/api.model.js

```diff
@@ -24,11 +23,34 @@ LEFT JOIN comments ON articles.article_id = comments.article_id
 GROUP BY articles.article_id
 ORDER BY articles.created_at DESC;
 `).then((result)=>{
-        console.log(result.rows)
         return result.rows
     })
     }
 
+    exports.retrieveCommentsById = (article_id) => {
+      return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
+        .then((result) => {
+          if (result.rows.length === 0) {
+            return Promise.reject({ status: 404, msg: 'article does not exist' });
+          }
+          return db.query(`
+            SELECT comment_id, 
+                   votes, 
+                   created_at, 
+                   author, 
+                   body, 
+                   article_id
+            FROM comments
+            WHERE article_id = $1
+            ORDER BY created_at DESC;
+          `, [article_id]);
+        })
+        .then((result) => {
+          console.log(result.rows)
+          return result.rows;
+        });
+    };  
```

Comment:
You could split this functionality out into two separate models, you could even take the top one further and make a generic 'check exists' function. Not necessary but something to ponder. 

You could also incorporate a Promise.all into this system, either in here or if you split this out into two models! 

Your logic here is sound though, just some suggestions. Don't forget to remove console logs!